tests/specs/controllers/CheckUploadedDocuments.test.js

```diff
@@ -0,0 +1,376 @@
+                        paymentStartPageUrl: 'stub_payment_url',
+                    },
+                },
+            },
+        };
+
+        resStub = {
+            redirect: sandbox.spy(),
+            serverError: sandbox.spy(),
+        };
+
+        sandbox.spy(sails.log, 'error');
+        sandbox.spy(sails.log, 'info');
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    describe('addDocsToDBHandler', () => {
+        it('should throw an error if no files are found', () => {
+            // when
+            reqStub.session.eApp.uploadedFileData = [];
+            CheckUploadedDocumentsController.addDocsToDBHandler(
+                reqStub,
+                resStub
+            );
+
+            // then
+            expect(resStub.serverError.calledOnce).to.be.true;
+            expect(sails.log.error.calledWith('No files uploaded')).to.be.true;
+        });
+
+        it('should upload files to the database if they exist', () => {
+            // when
+            const createUploadedDocumentsUrls = sandbox.stub(
+                UploadedDocumentUrls,
+                'create'
+            );
+            createUploadedDocumentsUrls.resolves();
+            CheckUploadedDocumentsController.addDocsToDBHandler(
+                reqStub,
+                resStub
+            );
+
+            // then
+            const firstCallArgs = {
+                application_id: 12345,
+                uploaded_url: 'aws_url_45678_test1.pdf',
+            };
+            const secondCallArgs = {
+                application_id: 12345,
+                uploaded_url: 'aws_url_45678_test2.pdf',
+            };
+
+            expect(createUploadedDocumentsUrls.callCount).to.equal(2);
+            expect(
+                createUploadedDocumentsUrls.getCall(0).args[0]
+            ).to.deep.equal(firstCallArgs);
+            expect(
+                createUploadedDocumentsUrls.getCall(1).args[0]
+            ).to.deep.equal(secondCallArgs);
+        });
+    });
+
+    describe('_dbColumnData', () => {
+        it('should throw an error if there is no appId', () => {
+            // when
+            reqStub.session.appId = null;
+            CheckUploadedDocumentsController.addDocsToDBHandler(
+                reqStub,
+                resStub
+            );
+
+            // then
+            expect(resStub.serverError.calledOnce).to.be.true;
+            expect(sails.log.error.calledWith('Missing applicaiton id')).to.be
+                .true;
+        });
+    });
+
+    describe('_checkDocumentCountAndPaymentDetails', () => {
+        it('should check document count and payment details', () => {
+            // when
+            sandbox.spy(
+                CheckUploadedDocumentsController,
+                '_checkDocumentCountInDB'
+            );
+            sandbox.spy(
+                CheckUploadedDocumentsController,
+                '_checkPaymentDetailsExistsInDB'
+            );
```

Comment:
I would do this, so then your calls further down are a bit cleaner/readable
```suggestion
            const checkCountSpy = sandbox.spy(
                CheckUploadedDocumentsController,
                '_checkDocumentCountInDB'
            );
            const checkPaymentSpy = sandbox.spy(
                CheckUploadedDocumentsController,
                '_checkPaymentDetailsExistsInDB'
            );
            //... 
            expect(checkCountSpy.calledWith(...)
```